docs/docs/rest-api/public/api/v2/apps.raml

```diff
@@ -0,0 +1,451 @@
+
+        - <code>apps.failures</code> Apps last failures are not embedded in
+        the response by default. This implies apps.lastTaskFailure, apps.tasks, apps.counts and
+        apps.deployments.
+
+        - <code>apps.taskStats</code> This exposes some task
+        statatistics in the JSON.
+
+      enum: [ apps.tasks, apps.count, apps.deployments, apps.lastTaskFailure, apps.failures, apps.taskStats ]
+      repeat: true
+      example: apps.tasks
+  responses:
+    200:
+      description: "The list of applications that match the defined filters"
+      body:
+        application/json:
+          example: !include examples/app_list.json
+
+###
+# update a list of applications
+#
+put:
+  description:
+    Change multiple applications either by upgrading existing ones or creating new ones.
+    If there is an update to an already running application, the application gets upgraded.
+    All instances of this application get replaced by the new version.
+    The order of dependencies will be applied correctly.
+    The upgradeStrategy defines the behaviour of the upgrade.
+
+    If the id of the application is not known, the application gets started.
+    The order of dependencies will be applied correctly.
+    It is possible to mix upgrades and installs.
+
+    If you have more complex scenarios with upgrades, use the groups endpoint.
+  is: [ secured ]
+  body:
+    application/json:
+      example: !include examples/apps_create.json
+  responses:
+    200:
+      description:
+        A deployment is started which has a unique deployment identifier.
+        The related deployment can be fetched from the /v2/deployments endpoint.
+        If the deployement is gone from the list of deployments, than this means it is finished.
+        As long as the deployment runs, the effect of that change operation is visible only partially.
+      body:
+        application/json:
+          example: !include examples/deployments_result.json
+    409:
+      description: There is an already deployed application with this name
+      body:
+        application/json:
+          example: |
+            {"message":"An app with id [/existing_app] already exists."}
+    400:
+      description: The application definition provided in the body is not valid.
+      body:
+        application/json:
+          example: |
+            {"message":"Invalid JSON","details":[{"path":"/id","errors":["error.expected.jsstring"]}]}
+    422:
+      description: The entity send can not be preocessed, since there are validation errors
+      body:
+        application/json:
+          example: |
+            {
+              "message": "Bean is not valid",
+              "errors": [
+                {
+                  "attribute": "upgradeStrategy.minimumHealthCapacity",
+                  "error": "is greater than 1"
+                }
+              ]
+            }
+
+
+
+
+###
+# Create application
+#
+post:
+  description: Create and start a new application.
```

Comment:
I have started to integrate this documentation into the json schema. It would be the natural place for raml documentation. Unfortunately there is no nice visualization. So my suggestion would be to add the model description to the introduction part of the documentation.